src/boot/efi/boot.c

```diff
@@ -1008,6 +1010,17 @@ static VOID config_defaults_load_from_file(Config *config, CHAR8 *content) {
                                 continue;
                         config->no_editor = !on;
                 }
+
+                if (strcmpa((CHAR8 *)"console_mode", key) == 0) {
+                        CHAR16 *s;
+
+                        s = stra_to_str(value);
+                        config->console_mode = Atoi(s);
```

Comment:
i am really not a fan of such conversions. Can't we do this with symbolic names instead? i.e. a short table that translates strings such as "80x25" or "80x50" into the right numbers?

Also, atoi eats parsing errors, and returns a signed, i doubt that's the best choice really

Reply:
@poettering not in that way, the mode resolutions are non standard and system specific, what we have is a mode identifier and creating a table does not work. Also, in this application, the user is not concerned by the exact screen resolution, it's expected to pick a mode identifier that makes the screen look ok.

The code is a copy-paste from other cases, but yes, it's ugly and missing error handling. We could change it to something else. Any hints with the UEFI library ? We might end up writing our own number parser.